cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/k8s/UserServiceProgramMain.java

```diff
@@ -0,0 +1,378 @@
+  private long maxStopSeconds;
+
+  /**
+   * Main entry point
+   */
+  public static void main(String[] args) throws Exception {
+    main(UserServiceProgramMain.class, args);
+  }
+
+  @Override
+  protected List<Module> getServiceModules() {
+    return Arrays.asList(
+      new NamespaceQueryAdminModule(),
+      new MessagingClientModule(),
+      new AuditModule(),
+      new AuthorizationEnforcementModule().getDistributedModules(),
+      new SecureStoreClientModule(),
+      new MetadataReaderWriterModules().getDistributedModules(),
+      new DistributedArtifactManagerModule(),
+      new AuthenticationContextModules().getProgramContainerModule(),
+      // Always use local table implementations, which use LevelDB.
+      // In K8s, there won't be HBase and the cdap-site should be set to use SQL store for StructuredTable.
+      new DataSetServiceModules().getStandaloneModules(),
+      // The Dataset set modules are only needed to satisfy dependency injection
+      new DataSetsModules().getDistributedModules(),
+      getDataFabricModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(ProgramStateWriter.class).to(MessagingProgramStateWriter.class);
+          bind(WorkflowStateWriter.class).to(MessagingWorkflowStateWriter.class);
+
+          // don't need to perform any impersonation from within user programs
+          bind(UGIProvider.class).to(CurrentUGIProvider.class).in(Scopes.SINGLETON);
+
+          bind(ServiceAnnouncer.class).toInstance(new ServiceAnnouncer() {
+            private final DiscoveryService discoveryService = masterEnv.getDiscoveryServiceSupplier().get();
+            private final String hostname = InetAddress.getLoopbackAddress().getCanonicalHostName();
+
+            @Override
+            public Cancellable announce(String serviceName, int port) {
+              return discoveryService.register(
+                ResolvingDiscoverable.of(new Discoverable(serviceName, new InetSocketAddress(hostname, port))));
+            }
+
+            @Override
+            public Cancellable announce(String serviceName, int port, byte[] payload) {
+              return discoveryService.register(
+                ResolvingDiscoverable.of(new Discoverable(serviceName, new InetSocketAddress(hostname, port),
+                                                          payload)));
+            }
+          });
+
+          bind(ExploreClient.class).to(UnsupportedExploreClient.class);
+          bind(OwnerAdmin.class).to(NoOpOwnerAdmin.class);
+        }
+      }
+    );
+  }
+
+  @Override
+  protected void addServices(Injector injector, List<? super com.google.common.util.concurrent.Service> services,
+                             List<? super AutoCloseable> closeableResources) {
+    services.add(injector.getInstance(MetricsCollectionService.class));
+  }
+
+  @Nullable
+  @Override
+  protected LoggingContext getLoggingContext() {
```

Comment:
shouldn't this return some logging context?

Reply:
Should return a `UserServiceLoggingContext` with system namespace.